src/hotspot/cpu/x86/c2_MacroAssembler_x86.cpp

```diff
@@ -763,25 +767,10 @@ void C2_MacroAssembler::fast_unlock(Register objReg, Register boxReg, Register t
   jccb  (Assembler::notZero, DONE_LABEL);
   movptr(boxReg, Address(tmpReg, OM_OFFSET_NO_MONITOR_VALUE_TAG(EntryList)));
   orptr(boxReg, Address(tmpReg, OM_OFFSET_NO_MONITOR_VALUE_TAG(cxq)));
-  jccb  (Assembler::notZero, CheckSucc);
+  jccb  (Assembler::notZero, DONE_LABEL);
```

Comment:
Because the code is arranged differently now. If we branched to CheckSucc, we would end up in the stack-locked path, which is incorrect. We really want to branch to DONE. That's also what we do in upstream JDK21 (I believe there has been some cleanup between 17 and 21 that fixed that).

Reply:
OK, so pessimistically if this is still a bug, we "just" enter the slow-path, right? If so, this looks acceptable.